lib/chat_api/reporting.ex

```diff
@@ -101,13 +121,38 @@ defmodule ChatApi.Reporting do
     |> compute_weekday_aggregates()
   end
 
+  def first_response_time_by_weekday(account_id, filters \\ %{}) do
+    Conversation
+    |> where(account_id: ^account_id)
+    |> where(^filter_where(filters))
+    |> where([conv], not is_nil(conv.first_replied_at))
+    |> average_grouped_by_date()
+    |> select_merge([m], %{day: fragment("to_char(date(?), 'Day')", m.inserted_at)})
+    |> Repo.all()
+    |> Enum.group_by(&String.trim(&1.day))
+    |> compute_average_weekday_aggregates()
+  end
+
   defp count_grouped_by_date(query, field \\ :inserted_at) do
     query
     |> group_by([r], fragment("date(?)", field(r, ^field)))
     |> select([r], %{date: fragment("date(?)", field(r, ^field)), count: count(r.id)})
     |> order_by([r], asc: fragment("date(?)", field(r, ^field)))
   end
 
+  # TODO some duplication here with group by date might be good to refactor
+  defp average_grouped_by_date(query, field \\ :inserted_at) do
+    query
+    |> group_by([r], fragment("date(?)", field(r, ^field)))
+    |> select([r], %{
+      date: fragment("date(?)", field(r, ^field)),
+      count: count(r.id),
+      # avg doesn't do anything but raises a grouping_error when I remove it...
```

Comment:
hmm that's a little concerning... would be nice to figure out the underlying issue here?